pkg/velerorunner/backup.go

```diff
@@ -178,3 +190,60 @@ func (t *Task) getReplicatedBackup() (*velero.Backup, error) {
 
 	return nil, nil
 }
+
+// Delete the running restic pod in velero namespace
+// This function is used to get around mount propogation requirements
+func (t *Task) bounceResticPod() error {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	// Set two vars to determine if we have bounced a pod and it is has become
+	// ready or if we need to delete the pod
+	var podDeleted = false
+	var podBounced = false
+	// Loop until pod has been deleted and is `Running`
+	for podBounced == false {
```

Comment:
Rather than having the entire pod bounce happen inside of a single reconcile like this, I think we need to figure out how to make the controller kick off a "bounce" action here, then exit the reconcile to wait for another action, then come back and check if the bounce was completed successfully before taking further action.

I think we can store some state in the `status` of the MigMigration to help with this, e.g. the podBounced and podDeleted vars could be moved there (although they would likely be renamed to be a bit more descriptive out of context, e.g. resticPodDeleted, resticPodBounced).

The challenging part here will be detecting when the restic pod state changes. I'm not sure if it's possible to detect this using a watch. We can look into it. Another option would be to exit with a requeue set for 10 seconds in the future or so.

What I'm trying to avoid here is the controller getting stuck inside of this loop, since MigStage / MigMigration only have one worker each. No other work can be done by these controller while they are waiting on this process to finish.

Reply:
Agreed with @djwhatle.

Reply:
Done, now we are using the status field to monitor the restic restart state